src/js/testem-component.js

```diff
@@ -0,0 +1,531 @@
+            args:     ["http://localhost:%port", { port: "{that}.options.coveragePort" }]
+        }
+    },
+    coverageDir: {
+        expander: {
+            funcName: "gpii.testem.generateUniqueDirName",
+            args:     [os.tmpdir(), "coverage", "{that}.id"] // basePath, prefix, suffix
+        }
+    },
+    reportsDir: {
+        expander: {
+            funcName: "gpii.testem.generateUniqueDirName",
+            args:     [os.tmpdir(), "reports", "{that}.id"] // basePath, prefix, suffix
+        }
+    },
+    testemDir: {
+        expander: {
+            funcName: "gpii.testem.generateUniqueDirName",
+            args:     [os.tmpdir(), "user_data_dir", "{that}.id"] // basePath, prefix, suffix
+        }
+    },
+    instrumentSource: true,
+    // Code to be served must live under the cwd in order to work with Testem.
+    instrumentedSourceDir: {
+        expander: {
+            funcName: "gpii.testem.resolveSafely",
+            args:     ["{that}.options.testemOptions.cwd", "instrumented"]
+        }
+    },
+    generateCoverageReport: true,
+    sourceDirs: [],
+    testPages:   [],
+    serveDirs:  [],
+    members: {
+        generatedOptions: {
+            routes: {}
+        }
+    },
+    events: {
+        initialCleanupComplete: null,
+        constructFixtures: null,
+        sourceInstrumented: null,
+        safeToConstructFixtures: {
+            events: {
+                initialCleanupComplete: "initialCleanupComplete",
+                constructFixtures:      "constructFixtures",
+                sourceInstrumented:     "sourceInstrumented"
+            }
+        },
+        onExpressStarted: null,
+        onFixturesConstructed: {
+            events: {
+                onExpressStarted: "onExpressStarted"
+            }
+        },
+        stopFixtures: null,
+        onExpressStopped: null,
+        onFixturesStopped: {
+            events: {
+                onExpressStopped: "onExpressStopped"
+            }
+        },
+        createReports: null,
+        onCoverageReportComplete: null,
+        onReportsComplete: {
+            events: {
+                onCoverageReportComplete: "onCoverageReportComplete"
+            }
+        },
+        finalCleanup: null,
+        finalCleanupComplete: null
+    },
+    testemOptions: {
+        framework:   "qunit",
+        report_file: {
+            expander: {
+                funcName: "gpii.testem.resolveSafely",
+                args:     ["{that}.options.reportsDir", "report.tap"]
+            }
+        },
+        cwd: process.cwd(),
+        user_data_dir: "{that}.options.testemDir",
+        on_start: "{that}.handleTestemStart",
+        on_exit:  "{that}.handleTestemExit",
+        test_page: "{that}.options.testPages",
+        proxies: {
+            "/coverage": {
+                "target": "{that}.options.coverageUrl"
+            }
+        }
+    },
+    invokers: {
+        "handleTestemStart": {
```

Comment:
I have I hope addressed this comment in spirit.  The latest code uses `fluid.promise.makeTransformer` to fire two "pseudo events" when Testem starts up and shuts down.  Namespaced and prioritised listeners can register themselves to respond, and are executed in sequence.

Although I had originally considered storing the Testem callbacks and referring to those in IoC references, I ended up adding the Testem callback as both the `onReject` and `onResolve` handler for the sequence returned by `fluid.promise.makeTransformer`.  As failing to call the callbacks is a fundamental error that results in a hung process, I felt it was appropriate to hard-code this bit of wiring as an implicit and immovable last step in the chain.